### PR TITLE
Add `unused-variable` example

### PR DESCRIPTION
**Pull request by harupy**
Co-authored-by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description

Refs #5953

Closes #XXXX

### CODE REVIEW

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Looks good thank you ! Could you use an example without using ``foo`` ? Generally we use fruit, animal school or cultural reference in examples. It makes the doc less "dry" and beginners do not understand "foo" / " bar".

File: doc/data/messages/u/unused-variable/bad.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+def foo():
+    x = 1
+    y = None  # [unused-variable]
+    print(x)
```
Comment (harupy) [inline doc/data/messages/u/unused-variable/bad.py:4]:
```python
def print_fruits():
    orange = "orange"
    apple = "apple"  # [unused-variable]
    print(orange)
```

Comment (harupy) [inline doc/data/messages/u/unused-variable/bad.py:4]:
```suggestion
def print_fruits():
    orange = "orange"
    apple = "apple"  # [unused-variable]
    print(orange)
```

Comment (Pierre-Sassoulas) [reply doc/data/messages/u/unused-variable/bad.py:4]: 👌

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**